src/ui/patch_hub/series_details.sh

```diff
@@ -1,102 +1,63 @@
 # @_target_patch_metadata: List with patches metadata
+# @show_bookmarked_patchset: If set, the patchset to be shown is a bookmarked
+#   one, i.e., the details will be fetch in local database not in
+#   `@_target_patch_metadata`.
 function show_series_details()
 {
-  local patch_index="$1"
-  local -n _target_patch_metadata="$2"
+  local series_index="$1"
+  local -n _target_patch_metadata
+  local show_bookmarked_patchset="$3"
   declare -A series
   local -a action_list
-  local -a check_statuses=('' '')
+  local -a check_statuses=()
   local patch_metadata
   local raw_series
   local message_box
 
-  # TODO: Add apply patch
-  action_list=('Bookmark' 'Download')
-
-  raw_series=${_target_patch_metadata["$patch_index"]}
-  parse_raw_series "${raw_series}" 'series'
-
-  patch_metadata=$(prettify_string 'Series:' "${series['patch_title']}")
-  patch_metadata+=$(prettify_string 'Author:' "${series['patch_author']}")
-  patch_metadata+=$(prettify_string 'Version:' "${series['patch_version']}")
-  patch_metadata+=$(prettify_string 'Patches:' "${series['total_patches']}")
-  message_box="$patch_metadata"
-
-  is_bookmarked "${raw_series}"
-  if [[ "$?" == 0 ]]; then
-    check_statuses[0]=1
-    # TODO: when we refine the 'Download' action, we should revise the set below
-    check_statuses[1]=1
-  fi
-
-  create_simple_checklist 'Patch(es) info and actions' "$message_box" 'action_list' 'check_statuses' 1
-  ret="$?"
-
-  case "$ret" in
-    0) # OK
-      IFS=' ' read -ra selected_options <<< "$menu_return_string"
-      for option in "${selected_options[@]}"; do
-        case "$option" in
-          'Bookmark')
-            handle_bookmark_action 'series' "$raw_series"
-            ;;
-          'Download')
-            handle_download_action 'series'
-            ;;
-        esac
-      done
-      ;;
+  if [[ -n "$show_bookmarked_patchset" ]]; then
+    action_list=('Unbookmark')
 
-    1) # Exit
-      handle_exit "$ret"
-      ;;
+    # The local bookmark database starting index is 1 and the index
+    # passed as argument starts at 0.
+    if [[ ! "${series_index}" =~ ^[0-9]+$ || "${series_index}" -lt 0 ]]; then
+      complain "Invalid value for series_index: '${series_index}'"
+      handle_exit 22 # EINVAL
+    fi
 
-    3) # Return
-      screen_sequence['SHOW_SCREEN']='show_new_patches_in_the_mailing_list'
-      screen_sequence['RETURNING']=1
-      ;;
-  esac
-}
+    ((series_index++))
 
-# Screen resposible for showing a specific bookmarked series details
-#
-# @series_index: Series index
-function show_bookmarked_series_details()
-{
-  local series_index="$1"
-  declare -A series
-  local -a action_list
-  local -a check_statuses=('')
-  local patch_metadata
-  local raw_series
-  local message_box
+    raw_series=$(get_bookmarked_series_by_index "${series_index}")
+  else
+    # TODO: Add apply patch
+    action_list=('Bookmark' 'Download')
 
-  action_list=('Unbookmark')
+    _target_patch_metadata="$2"
```

Comment:
Assigning the second parameter here is not intuitive and breaks our code convention. Could you rethink the logic in a way that the second parameter is assigned at the beginning of this function?